error/s2n_errno.c

```diff
@@ -286,6 +286,7 @@ static const char *no_such_error = "Internal s2n error";
     ERR_ENTRY(S2N_ERR_OSSL_PROVIDER, "Failed to load or unload an openssl provider") \
     ERR_ENTRY(S2N_ERR_CERT_OWNERSHIP, "The ownership of the certificate chain is incompatible with the operation") \
     ERR_ENTRY(S2N_ERR_INTERNAL_LIBCRYPTO_ERROR, "An internal error has occurred in the libcrypto API") \
+    ERR_ENTRY(S2N_ERR_KTLS, "An internal error while enabling kTLS") \
```

Comment:
nit:
```suggestion
    ERR_ENTRY(S2N_ERR_KTLS, "An internal error has occurred while enabling kTLS") \
```

Also, I imagine there might be more kTLS errors in the future. Would something like S2N_ERR_KTLS_ENABLE maybe be better, so it's more specific?

Reply:
Alot of the kTLS errors are actually going to be recoverable and even this one is captured and not returned. So I am more inclined to keep this generic for now and make it more specific once we have the usecase.